local-volume/provisioner/pkg/common/common.go

```diff
@@ -57,6 +64,8 @@ type MountConfig struct {
 	HostDir string `json:"hostDir"`
 	// The mount point of the hostpath volume
 	MountDir string `json:"mountDir"`
+	// The volumeType "block" or "file". If nil, the type defaults to "file"
+	VolumeType string `json:"volumeType"`
```

Comment:
It is nice that this can catch user errors, although I imagine, it would be difficult to make a mistake because as an admin you would need to do two entirely different commands?  For block devices, you need to symlink the device into the directory.  For fs volumes, you need to run a mount command into the directory.

With auto-detection, we could support a mix of both in the same directory (and storage class)